src/mixins/api-mixin.ts

```diff
@@ -0,0 +1,129 @@
+// Libraries
+import axios from '@/utils/axios-auth'
+import { Component, Mixins, Vue } from 'vue-property-decorator'
+import { State, Action, Getter } from 'vuex-class'
+
+// Interfaces
+import { StateModelIF, ActionBindingIF, GetterIF } from '@/interfaces'
+
+// Utils
+import { FilingTemplateMixin } from '@/mixins'
+
+/**
+ * Mixin that provides the integration with the legal api.
+ */
+@Component
+export default class ApiMixin extends Mixins(FilingTemplateMixin) {
```

Comment:
I am ok with having this as a mixin opposed to the service pattern. In that case the name needs to be more specific I think. We will be calling Name Requests API, Auth API etc.

Reply:
This is outdated now, the filingtemplate is its own mixin and is now built when the save filing is called and passes it to the api mixin